rules/msk_consume_in_module_test.go

```diff
@@ -0,0 +1,147 @@
+package rules
+
+import (
+	"testing"
+
+	"github.com/hashicorp/hcl/v2"
+	"github.com/stretchr/testify/require"
+	"github.com/terraform-linters/tflint-plugin-sdk/helper"
+)
+
+func Test_MskConsumeInModuleRule(t *testing.T) {
+	rule := &MskConsumeInModuleRule{}
+
+	for _, tc := range []struct {
+		name     string
+		files    map[string]string
+		expected helper.Issues
+	}{
+		{
+			name: "consuming from topic not in module",
+			files: map[string]string{
+				"file.tf": `
+module "consumer" {
+	consume_topics = ["some_topic"]
+}
+`,
+			},
+			expected: []*helper.Issue{
+				{
+					Rule:    rule,
+					Message: "'consume_topics' may only contain topics defined in the current module but 'some_topic' is not",
+					Range: hcl.Range{
+						Filename: "file.tf",
+						Start:    hcl.Pos{Line: 3, Column: 2},
+						End:      hcl.Pos{Line: 3, Column: 33},
+					},
+				},
+			},
+		},
+		{
+			name: "producing from topic not in module",
+			files: map[string]string{
+				"file.tf": `
+module "consumer" {
+	produce_topics = ["some_topic"]
+}
+`,
+			},
+			expected: []*helper.Issue{
+				{
+					Rule:    rule,
+					Message: "'produce_topics' may only contain topics defined in the current module but 'some_topic' is not",
+					Range: hcl.Range{
+						Filename: "file.tf",
+						Start:    hcl.Pos{Line: 3, Column: 2},
+						End:      hcl.Pos{Line: 3, Column: 33},
+					},
+				},
+			},
+		},
+		{
+			name: "producing and consuming from topic not in module",
+			files: map[string]string{
+				"file.tf": `
+module "consumer" {
+	consume_topics = ["some_topic"]
+	produce_topics = ["some_topic"]
+}
+`,
+			},
+			expected: []*helper.Issue{
+				{
+					Rule:    rule,
+					Message: "'consume_topics' may only contain topics defined in the current module but 'some_topic' is not",
+					Range: hcl.Range{
+						Filename: "file.tf",
+						Start:    hcl.Pos{Line: 3, Column: 2},
+						End:      hcl.Pos{Line: 3, Column: 33},
+					},
+				},
+				{
+					Rule:    rule,
+					Message: "'produce_topics' may only contain topics defined in the current module but 'some_topic' is not",
+					Range: hcl.Range{
+						Filename: "file.tf",
+						Start:    hcl.Pos{Line: 4, Column: 2},
+						End:      hcl.Pos{Line: 4, Column: 33},
+					},
+				},
+			},
+		},
+		{
+			name: "external topic defined outside of consumer/producer",
+			files: map[string]string{
+				"file.tf": `
+module "some_other_module" {
+	kafka_stuff = ["external_topic"]
+}
+`,
+			},
+			expected: []*helper.Issue{},
+		},
+		{
+			name: "all topics defined in file",
+			files: map[string]string{
+				"file.tf": `
+resource "kafka_topic" "first_topic" {
```

Comment:
Would be nice to define these topics in separate files, to make sure that case is covered too.

Reply:
> Would be nice to define these topics in separate files, to make sure that case is covered too.

:+1: 6c86721